### PR TITLE
Revert "apm: rm deprecated peer_service_aggregation config"

### PR DESCRIPTION
**Pull request by pkalmakis**
Reverts DataDog/datadog-agent#23284

This config is still in use - let's not delete this until we have ensured we migrate everyone off

### CODE REVIEW

**Review by ajgajg1134 — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1282,6 +1282,11 @@ api_key:
   ## If you are sending OTel traces and do not want stats computed by span kind, you need to disable this flag and remove the "enable_otlp_compute_top_level_by_span_kind" APM feature if present.
   # compute_stats_by_span_kind: false
 
+  ## @param peer_service_aggregation - bool - default: false
+  ## @env DD_APM_PEER_SERVICE_AGGREGATION - bool - default: false
+  ## DEPRECATED - please use `peer_tags_aggregation` instead.
```
Comment (maycmlee) [inline pkg/config/config_template.yaml:1287]:
```suggestion
  ## DEPRECATED - use `peer_tags_aggregation` instead.
```

**Review by maycmlee — APPROVED** [review]
Small edit, but approving

**Review by liustanley — APPROVED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (pkalmakis) [reply pkg/config/config_template.yaml:1287]: skipping this as this PR is simply a revert of another change

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**